lib/node/server.ml

```diff
@@ -22,32 +21,80 @@ let handle_response request_table res =
    4. Grab the next request if it exists and send it to the message handler along with the
      node's state
    5. Send the encoded response from the message handler to the requester *)
-let run node router msg_handler =
-  let rec server () =
-    let%lwt message = Client.recv_next node in
-    let%lwt () = route node router message in
-
-    let%lwt () =
-      match%lwt Inbox.next !node.inbox Message.Failure_detection with
-      | Some message -> Failure_detector.handle_message node message
-      | None -> Lwt.return () in
-
-    let%lwt () = Failure_detector.failure_detection node in
-
-    let%lwt next_response = Inbox.next !node.inbox Message.Response in
-    let _ = handle_response !node.request_table next_response in
-
-    let%lwt request = Inbox.next !node.inbox Message.Request in
+let rec run last node preprocessor msg_handler =
```

Comment:
I think this function definitively needs some comments inside its scope. I found it pretty hard to read and understand :/

Reply:
Yeah, that's definitely true.